notebook/static/notebook/js/notebook.js

```diff
@@ -325,6 +325,11 @@ define(function (require) {
         var x = new commandpalette.CommandPalette(this);
     };
 
+    Notebook.prototype.show_search_and_replace = function(event) {
+        var kbd = this.keyboard_manager;
+        kbd.actions.call('ipython.search-and-replace-dialog', event, kbd);
+    }
+
```

Comment:
That feel to me like an extra step to me, directly do `action.call(...)` in `menubar.js` and no need to pass kbd at the end.